src/opm/parser/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -1211,6 +1215,59 @@ bool FieldProps::tran_active(const std::string& keyword) const {
     return calculator != this->tran.end() && calculator->second.size() > 0;
 }
 
+void FieldProps::apply_numerical_aquifers(const NumericalAquifers& numerical_aquifers) {
+    // TODO: ideally, we should also update the cell_depth, but it is not used for later
+    // in the simulator. So we do not update it here.
+    // Maybe also cell volume? Not sure whether it is used later
+    auto& porv_data = this->init_get<double>("PORV").data;
+    auto& satnum_data = this->int_data["SATNUM"].data;
+    auto& pvtnum_data = this->int_data["PVTNUM"].data;
+
+    const auto& aqu_cell_props = numerical_aquifers.aquiferCellProps();
+    for (const auto& [global_index, cellprop] : aqu_cell_props) {
+        const size_t active_index = this->grid_ptr->activeIndex(global_index);
+        porv_data[active_index] = cellprop.pore_volume;
+        satnum_data[active_index] = cellprop.satnum;
+        pvtnum_data[active_index] = cellprop.pvtnum;
+    }
+
+    this->updateTransWithNumericalAquifer(numerical_aquifers);
+}
+
+void FieldProps::updateTransWithNumericalAquifer(const NumericalAquifers& numerical_aquifers) {
+    const std::array<std::set<size_t>, 3> trans_to_remove = numerical_aquifers.transToRemove(*(this->grid_ptr));
+    std::array<std::vector<Box::cell_index>, 3> index_lists;
+    for (int i = 0; i < 3; ++i) {
+        size_t num = 0;
+        for (const auto& elem : trans_to_remove[i]) {
+            const size_t active_index = this->grid_ptr->activeIndex(elem);
+            index_lists[i].emplace_back(elem, active_index, num);
+            num++;
+        }
+    }
+
+    const std::array<std::string, 3> trans_string {"TRANX", "TRANY", "TRANZ"};
+    for (int i = 0; i < 3; ++i) {
+        const std::string& target_kw = trans_string[i];
+        const std::vector<Box::cell_index>& single_index_list = index_lists[i];
+        auto tran_iter = this->tran.find(target_kw);
+        if (tran_iter == this->tran.end()) {
+            const std::string msg = trans_string[i] + " TranCalculator could not be found when applying numerical aquifer";
+            throw std::logic_error(msg);
+        }
+        assert(tran_iter != this->tran.end());
```

Comment:
This `assert()` should be removed - the condition is tested with an `if` three lines up!

Reply:
Yes. Thanks for pointing out.